perceval/utils/logging.py

```diff
@@ -0,0 +1,249 @@
+        exqalibur_logging.set_level(_LEVEL_CONVERTER[level], _CHANNEL_CONVERTER[channel])
+
+    @staticmethod
+    def debug(msg: str, channel: str = "user"):
+        exqalibur_logging.debug(str(msg), _CHANNEL_CONVERTER[channel])
+
+    @staticmethod
+    def info(msg: str, channel: str = "user"):
+        exqalibur_logging.info(str(msg), _CHANNEL_CONVERTER[channel])
+
+    @staticmethod
+    def warn(msg: str, channel: str = "user"):
+        exqalibur_logging.warn(str(msg), _CHANNEL_CONVERTER[channel])
+
+    @staticmethod
+    def _format_exception(exc_info=None) -> str:
+        if not exc_info:
+            return ""
+        return ': '+' '.join(traceback.format_exception(exc_info[0], exc_info[1], exc_info[2])).replace("\n", " ").replace("    ", " ").replace("  ", ", ")
+
+    @staticmethod
+    def error(msg: str, channel: str = "user", exc_info=None):
+        msg = str(msg)
+        if exc_info:
+            msg += ExqaliburLogger._format_exception(exc_info)
+            traceback.print_exception(exc_info[0], exc_info[1], exc_info[2])
+        exqalibur_logging.error(str(msg), _CHANNEL_CONVERTER[channel])
+
+    @staticmethod
+    def critical(msg: str, channel: str = "user", exc_info=None):
+        msg = str(msg)
+        if exc_info:
+            msg += ExqaliburLogger._format_exception(exc_info)
+            traceback.print_exception(exc_info[0], exc_info[1], exc_info[2])
+        exqalibur_logging.critical(str(msg), _CHANNEL_CONVERTER[channel])
+
+
+class PythonLogger():
+    def __init__(self) -> None:
+        self._logger = python_logging.getLogger()
+        self._logger.addFilter(CONFIG.message_has_to_be_logged)
+
+    def enable_file(self):
+        python_logging.warn("This method have no effect. Use module logging to configure python logger")
+
+    def disable_file(self):
+        python_logging.warn("This method have no effect. Use module logging to configure python logger")
+
+    def set_level(self, level: int, channel: str = "user"):
+        CONFIG.set_level(level, channel)
+
+    def save_config(self):
+        CONFIG.save()
+
+    def reset_config(self):
+        CONFIG.set_to_default_config()
+
+    def debug(self, msg: str, channel: str = "user"):
+        self._logger.debug(f"[debug] {msg}", extra={"channel": channel})
+
+    def info(self, msg: str, channel: str = "user"):
+        self._logger.info(f"[info] {msg}", extra={"channel": channel})
+
+    def warn(self, msg: str, channel: str = "user"):
+        self._logger.warn(f"[warning] {msg}", extra={"channel": channel})
+
+    def error(self, msg: str, channel: str = "user", exc_info=None):
+        self._logger.error(f"[error] {msg}", exc_info=exc_info, extra={
+                           "channel": channel})
+
+    def critical(self, msg: str, channel: str = "user", exc_info=None):
+        self._logger.critical(f"[critical] {msg}", exc_info=exc_info, extra={
+                              "channel": channel})
+
+
+ExqaliburLogger.initialize()
```

Comment:
Instead of calling `ExqaliburLogger.initialize()` everytime, why wouldn't you call a `_initialize()` private function checking what's in the config. Maybe the exqalibur logger isn't even going to need to be initialized, so don't do it in that case.